### DISCUSSION TITLE
Texture generation at a particular angle/direction/plane

### DISCUSSION THREAD

**Discussion by makeabhishek** [mtex-toolbox/mtex#1141] [Ask Anything]
Hi MTEX team,
I am trying to generate a random texture from odf. I can generate random texture as shown in the below figure using the code 
`odf = 0.5*fibreODF(Miller(0,0,0,1,csHex),xvector,'halfwidth',20*degree) + 0.5*fibreODF(Miller(0,0,0,1,csHex),zvector,'halfwidth',20*degree);`
![Picture1](https://user-images.githubusercontent.com/47937684/126427347-a420e04e-1101-4393-8d50-32b20b1207ba.png)

#### 3. Final question, how can we create a texture something like an elliptical shape rather than a complete circular pattern, which can be generated by `odf = fibreODF(Miller(0,0,0,1,csHex),zvector,'halfwidth',20*degree)` .

Thanks 
Regards
Abhishek

**Comment by kilir** [comment]
Hi,
wrt 1: you can rotate your odf
wrt 2: plotPDF takes care of symmetrically equivalent directions, so the labelling should indeed be {} resp. <>.
wrt 3: have a look at BinghamODF

Hope that helps.
Cheers, 
Rüdiger

**Reply by makeabhishek** [reply]
Thanks, Rüdiger for the answers.

wrt 1: Does the rotation of odf means to give rotation to the orientation of Euler like,

`ori1 = orientation.byEuler(60*degree,40*degree,60*degree,cs);`

wrt 2: yes, for demonstration, how to represent the polar plot in { } instead of ().

Regards
Abhi

**Reply by kilir** [reply]
Hi,
no, it means you rotate the odf with respect to specimen coordinates such that e.g. pole figures (which are in specimen coordinates) may appear differently as a result of this rotation. Similar to rotating an orientation, rotations can be defined by means of Euler angles, but also by means of axis/angle or a rotation matrix.

Cheers,
Rüdiger

**Reply by makeabhishek** [reply]
Thanks Rüdiger.
Now I used 
`rot = rotation.byAxisAngle(yvector,15*degree);`

Thanks for your support.
Abhishek
